frontend/src/sorted_table.tsx

```diff
@@ -0,0 +1,311 @@
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const handleRequestSort = (event: React.MouseEvent<unknown>, property: any) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const createSortHandler = (property: any) => (event: React.MouseEvent<unknown>) => {
+    handleRequestSort(event, property);
+  };
+
+  return <>
+    <div className={props.className ?? styles.table} style={props.style ?? props.style}>
+      { data
+        ? <>
+            <TableContainer>
+              <Table
+                aria-labelledby="tableTitle"
+                size={'medium'}
+                aria-label="enhanced table"
+              >
+                <TableHead>
+                  <TableRow>
+                    {visibleColumns.map((column) => (
+                      <TableCell
+                        key={column.field}
+                        align={column.numeric ? 'right' : 'left'}
+                        padding={'default'}
+                        sortDirection={orderBy === column.field ? order : false}
+                      >
+                        <TableSortLabel
+                          active={orderBy === column.field}
+                          direction={orderBy === column.field ? order : 'asc'}
+                          onClick={createSortHandler(column.field)}
+                        >
+                          {column.label}
+                          {orderBy === column.field ? (
+                            <span className={styles.sorted}>
+                              {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                            </span>
+                          ) : null}
+                        </TableSortLabel>
+                      </TableCell>
+                    ))}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {sortedData
+                    .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                    .map((row, index) => {
+                      return (
+                        <TableRow key={row.tableId}
+                          hover
+                          selected={selected == row.tableId}
+                          onClick={props.onClick
+                                   ? () => {
+                                     if (selected == row.tableId) {
+                                       props.onClick(null)
+                                       setSelected(null)
+                                     } else {
+                                       props.onClick(row)
+                                       setSelected(row.tableId)
+                                     }
+                                   }
+                                   : () => {}}
```

Comment:
```suggestion
                          onClick={props.onClick && () => {
                                     if (selected == row.tableId) {
                                       props.onClick(null)
                                       setSelected(null)
                                     } else {
                                       props.onClick(row)
                                       setSelected(row.tableId)
                                     }
                                   }}
```
I just use `&&` here, it's ok to pass undefined to event handlers